src/git.rs

```diff
@@ -212,7 +212,9 @@ impl Repository {
             callbacks.push_update_reference(|refname, status| {
                 assert_eq!(refname, "refs/heads/master");
                 if let Some(s) = status {
-                    ref_status = Err(format!("failed to push a ref: {}", s).into())
+                    ref_status = Some(Err(format!("failed to push a ref: {}", s).into()))
+                } else {
+                    ref_status = Some(Ok(()));
```

Comment:
I think this could still mask some errors.

[The documentation of git2](https://docs.rs/git2/0.11.0/git2/struct.RemoteCallbacks.html#method.push_update_reference) says *"Set a callback to get invoked for **each** updated reference on a push."* (emphasis mine): if the first ref has an error while the second ref was pushed successfully, this would override the error of the first ref.